test/tic_tac_toe_test.exs

```diff
@@ -1,7 +1,29 @@
 defmodule TicTacToeTest do
   use ExUnit.Case
 
-  test "welcomes the user to Tic-Tac-Toe" do
-    assert TicTacToe.run() == nil
+  defmodule FakePresenter do
+    def present(board) do
+      board_representation =
+        Enum.map(board, fn cell_value -> cell_value || "-" end)
+
+      Enum.join(board_representation)
```

Comment:
FWIW you can pipe here!

```suggestion
      board
      |> Enum.map(fn cell_value -> cell_value || "-" end)
      |> Enum.join()
```

Just a suggestion, no change necessary.

Reply:
I always miss these pipe opportunities